src/compiler/checker.ts

```diff
@@ -31515,6 +31516,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 }
             }
 
+            if (getOutermostDecoratedClass(right)) {

```

Comment:
I think the right name is `hasClassAncestorOutsideOfDecorator` or something like that.

Also, can you just use the new check inside of `checkGrammarPrivateIdentifierExpression`?